File: Folder8/Folder8.2/no4.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>
// #include <string.h>
using namespace std;

int has4(long int);
void go(long int);

void go(long int num) {
    char q[100];
    int t = sprintf(q, "%ld", num);
    // cout << t;
    string ans = "";
    string ori = "";
    for (int i = 0; i < t; ++i) {
        // cout << q[i] << " ";
        if (q[i] == '4') {
            ans+='1';
            ori+='3';
        } else {
            ans+='0';
            ori+=q[i];
        }
    }
    stringstream x(ans);
    int y = 0, z = 0;
    x >> y;
    stringstream j(ori);
    j >> z;
    cout << y << " " << z;
}

int main()
{
    int x, num;
    cin >> x;
    for (int i = 0; i < x; ++i)
    {
        cin >> num;
        cout << "Case #" << i+1 << ": ";
        go(num);
        cout << "\n";
    }
}

File: Folder8/Folder8.2/no4.js
const has4 = (int) => {
    for (let i = 0; i < int.toString(); ++i) {
        if (int.toString()[i] == "4") {
            return true;
        }
    }
    return false;
};
const go = (num) => {
    let x = has4(num), y = num, z = 0;
    while (x) {
        --y;
        ++z;
        x = has4(y) && has4(z);
    }
    return [y, z];
};

const cin = process.stdin;

// Set input character encoding.
cin.setEncoding('utf-8');

// Prompt user to input data in console.
console.log("Please input text in command line.");

// When user input data and click enter key.
cin.on('data', function (data) {
    if (data === 'exit\n') {
        process.exit();
    } else {
        console.log(go(Number(data)));
        process.exit();
    }
});